### PR TITLE
 Implemented EndOfSentenceFormat to support urls as last argument

### PR DESCRIPTION
**Pull request by schalkms**
reference #768

### CODE REVIEW

File: detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/EndOfSentenceFormat.kt [hunk 1]
```diff
@@ -24,42 +25,49 @@ class KDocStyle(config: Config = Config.empty) : MultiRule() {
...
 			Severity.Maintainability,
-			"The first sentence in a KDoc comment should end with correct punctuation.")
+			"The first sentence in a KDoc comment should end with correct punctuation.",
+			Debt.FIVE_MINS)
 
 	private val endOfSentenceFormat =
 			Regex(valueOrDefault(END_OF_SENTENCE_FORMAT, "([.?!][ \\t\\n\\r\\f<])|([.?!]\$)"))
 
+	// https://stackoverflow.com/questions/3809401/what-is-a-good-regular-expression-to-match-a-url
+	private val urlFormat = Regex("[-a-zA-Z0-9@:%._+~#=]{2,256}\\.[a-z]{2,6}\\b([-a-zA-Z0-9@:%_+.~#?&//=]*)")
```
Comment (vanniktech) [inline detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/EndOfSentenceFormat.kt]: We could leave the job to Javas URL class which throws an exception upon instantiation of an invalid url. But regex should work too

Comment (schalkms) [reply detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/EndOfSentenceFormat.kt]: I also thought about that. Javas URL class needs a URL with protocol definition though.
`www.google.com` would be invalid, but detekt could just append a protocol then.
Furthermore links to OS-dependent files and directories wouldn't be supported then.

File: detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/EndOfSentenceFormat.kt [hunk 2]
```diff
@@ -24,42 +25,49 @@ class KDocStyle(config: Config = Config.empty) : MultiRule() {
...
+				report(CodeSmell(issue, Entity.from(declaration),
+						"The first sentence of this KDoc does not end with the correct punctuation."))
 			}
 		}
 	}
 
 	private fun String.startsWithHtmlTag() = startsWith("<") && contains(htmlTag)
 
+	private fun lastArgumentMatchesUrl(text: String): Boolean {
+		val arguments = text.trimEnd().split(Regex("\\s+"))
```
Comment (arturbosch) [inline detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/EndOfSentenceFormat.kt]: Please extract this refex and url regex into top level variables and reuse them. Regex creation and the `matches()` function are really expensive when doing them everytime.

File: detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/EndOfSentenceFormat.kt (hunk 1 above)
Comment (arturbosch) [reply detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/EndOfSentenceFormat.kt]: I think URL is cheaper to use performance-wise (does not use a regex). But I think it does not handle all cases. 
Lets just use the url class and change it later if a user finds an issue with that?

File: detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/EndOfSentenceFormat.kt (hunk 2 above)
Comment (schalkms) [reply detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/documentation/EndOfSentenceFormat.kt]: Thanks! In all the hurry I forgot to extract this regex to a property.

**Pull request merged**